pulsar-common/src/main/java/org/apache/pulsar/common/util/ThreadDumpUtil.java

```diff
@@ -39,51 +39,73 @@
 public class ThreadDumpUtil {
     private static final String INDENT = "    ";
 
+    public interface DiagnosticProcessor {
```

Comment:
Can you explains a little about the modification in this class?

Reply:
The ThreadDumpUtil. buildThreadDiagnosticString() returns a huge string that has the stack traces of all the threads as well as the lock info and deadlock info. The size of the buffer could be several kilobytes to a megabyte. Some third-party logging services have limitations on the maximum size of a message that they can ingest. The idea is to break the log output to several messages so that each individual message can be easily ingested.

In the changes that I implemented, the default implementation is left as-is. But the client code has the option to register a callback for individual thread stack traces and lock info. The callback can choose to either build one huge buffer that has everything or break the individual stack traces and log it.